homeassistant/components/hassio.py

```diff
@@ -60,13 +73,39 @@
     vol.Required(ATTR_INPUT): vol.Any(dict, cv.string)
 })
 
+SCHEMA_SNAPSHOT_FULL = vol.Schema({
+    vol.Optional(ATTR_NAME): cv.string,
+})
+
+SCHEMA_SNAPSHOT_FULL = SCHEMA_SNAPSHOT_FULL.extend({
+    vol.Optional(ATTR_FOLDERS): vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(ATTR_ADDONS): vol.All(cv.ensure_list, [cv.string]),
+})
+
+SCHEMA_RESTORE_FULL = vol.Schema({
+    vol.Required(ATTR_SNAPSHOT): cv.slug,
+})
+
+SCHEMA_RESTORE_PARTIAL = vol.Schema({
+    vol.Optional(ATTR_HOMEASSISTANT): cv.boolean,
+    vol.Optional(ATTR_FOLDERS): vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(ATTR_ADDONS): vol.All(cv.ensure_list, [cv.string]),
+})
+
 MAP_SERVICE_API = {
-    SERVICE_ADDON_START: ('/addons/{addon}/start', SCHEMA_ADDON),
-    SERVICE_ADDON_STOP: ('/addons/{addon}/stop', SCHEMA_ADDON),
-    SERVICE_ADDON_RESTART: ('/addons/{addon}/restart', SCHEMA_ADDON),
-    SERVICE_ADDON_STDIN: ('/addons/{addon}/stdin', SCHEMA_ADDON_STDIN),
-    SERVICE_HOST_SHUTDOWN: ('/host/shutdown', None),
-    SERVICE_HOST_REBOOT: ('/host/reboot', None),
+    SERVICE_ADDON_START: ('/addons/{addon}/start', SCHEMA_ADDON, 60),
+    SERVICE_ADDON_STOP: ('/addons/{addon}/stop', SCHEMA_ADDON, 60),
+    SERVICE_ADDON_RESTART: ('/addons/{addon}/restart', SCHEMA_ADDON, 60),
+    SERVICE_ADDON_STDIN: ('/addons/{addon}/stdin', SCHEMA_ADDON_STDIN, 60),
+    SERVICE_HOST_SHUTDOWN: ('/host/shutdown', None, 60),
```

Comment:
This needs a schema of accepting nothing. Otherwise one can send any data and it will be send to the Hass.io API, potentially being a security issue.